### PR TITLE
Updates sidecar setup and run

### LINKED ISSUE #198: Create setup for sidecar
The origin of this issue is that the sidecar could import simcore-sdk but not its dependencies because they were actually *not* installed. This issue blocks #186. In order to solve it properly we need to fulfill:

- [x] Missing ``setup.py``
- [x] Missing requirement-dev.txt
- [ ] Non-root user in docker. 
    - devel target will not create eggs as root and in production :
```bash
usr/local/lib/python3.6/site-packages/celery/platforms.py:795: RuntimeWarning: You're running the worker with superuser privileges: this is
sidecar_1    | absolutely not recommended!
```
- [x] ~venv in docker~ wont' do. For this case, I agree that docker provides desired [isolation](https://stackoverflow.com/questions/48561981/activate-python-virtualenv-in-dockerfile)
- [x] rearrange package structure: ``main.py``, ``tasks.py``,  ``celery.py`` and ``utils.py``

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

Upgrades ``sidecar`` setup and module structure and fixes some issues.

- [x] Missing setup.py
- [x] Missing requirement-dev.txt
- [x] Non-root user in docker.
- [x] venv in docker
- [x] rearrange package structure: ``main.py``, ``tasks.py``,  ``celery.py`` and ``utils.py``
- [x] entrypoint script to start as ``root`` and run as ``scu``

Extra fixes on ``webserver``
- [x] retreived config files as resources
- [x] modified Dockerfile: should replicate the way we test in source
    - keeps same relative folder structure
    - all 3erd, 2nd and main packages install in a venv
    - runs as a non-root user
    - boot script

## Related issue number
- closes #198 

## Checklist

- [x] I think the code is well written
-  ~Unit tests for the changes exist~
- ~ Documentation reflects the changes~
- [x] If you design a new module, add your user to .github/CODEOWNERS

@mguidon, @sanderegg

### CODE REVIEW

File: services/docker-compose.devel.yml [hunk 1]
```diff
@@ -18,18 +18,22 @@ services:
   webserver:
     image: services_webserver:dev
     build:
+      args:
+        - HOST_GID_ARG=${HOST_GID:?Undefined host gid}
```
Comment (sanderegg) [inline services/docker-compose.devel.yml]: What does this syntax mean?

File: services/sidecar/Dockerfile [hunk 2]
```diff
@@ -1,33 +1,98 @@
...
-    && pip install -r requirements.txt \
-    && pip list --format=columns
+ENV HOME /home/scu
+ENV PIP  /home/scu/.venv/bin/pip3
 
 EXPOSE 8000
+VOLUME /home/scu/input
+VOLUME /home/scu/output
+VOLUME /home/scu/log
+#VOLUME /var/run/docker.sock DO NOT EXPOSE .. fails!
```
Comment (sanderegg) [inline services/sidecar/Dockerfile]: probably not needed here?

File: services/web/server/src/server/settings/config.py [hunk 3]
```diff
@@ -54,17 +56,19 @@
     T.Key("director"): _DIRECTOR_SCHEMA,
     T.Key("postgres"): db.CONFIG_SCHEMA,
     T.Key("rabbit"): rabbit.CONFIG_SCHEMA,
-    T.Key("s3"): s3.CONFIG_SCHEMA,
-    T.Key("cs_s4l"): _CS_S4L_SCHEMA
+    T.Key("s3"): s3.CONFIG_SCHEMA
+    #T.Key("cs_s4l"): _CS_S4L_SCHEMA
```
Comment (sanderegg) [inline services/web/server/src/server/settings/config.py]: can be safely removed

File: services/docker-compose.devel.yml (hunk 1 above)
Comment (pcrespov) [reply services/docker-compose.devel.yml]: RTFM? :-b

See the end of the section https://docs.docker.com/compose/compose-file/#variable-substitution

**Comment by sanderegg** [comment]
so this does not work on Windows. There is no docker group so the anyone mounting the docker socket as standard user is not able to access the docker engine.

**Pull request merged**